### PR TITLE
Add waypoints API

### PR DESCRIPTION
**Pull request by msg7086**

### CODE REVIEW

**Comment by tpendragon** [comment]
Closer. I think a more accurate API is a nested resource for images. So images has its own controller, but it's nested within waypoints. If this conflicts with what you're thinking just leave a note.

File: app/controllers/waypoints_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class WaypointsController < ApplicationController
+  respond_to :json
+
+  def index
+    visited_waypoints = session[:visited] || []
+    locations = Location.all
```
Comment (tpendragon) [inline app/controllers/waypoints_controller.rb:6]: Why does the controller for Waypoints manage Locations? I realize they're the same thing, but terminology needs to be consistent.

Comment (msg7086) [inline app/controllers/waypoints_controller.rb:6]: How to solve the routing conflicts between admin::locations and locations?

Comment (tpendragon) [inline app/controllers/waypoints_controller.rb:6]: What routing conflicts? Your admin::locations is namespaced - you have admin_locations_path and locations_path.

Comment (tpendragon) [inline app/controllers/waypoints_controller.rb:6]: Oh, you don't. Set your admin namespace up like this: https://github.com/osulp/Room-Reservation/blob/develop/config/routes.rb#L16 and you'll get admin_locations_path.

**Pull request closed without merging**